File: app/src/main/java/com/example/rickmortymvvm/util/observer/MutableObservableImpl.java
package com.example.rickmortymvvm.util.observer;


import com.example.rickmortymvvm.intrefaces.Observer;
import com.example.rickmortymvvm.list.viewmodel.PresentationCharacterListAction;
import com.example.rickmortymvvm.list.viewmodel.PresentationCharacterListState;

public class MutableObservableImpl implements MutableObservable {
    private Observer observer;

    @Override
    public void update(PresentationCharacterListState state) {
        observer.notify(state);
    }

    @Override
    public void update(PresentationCharacterListAction action) {
        observer.notify(action);
    }

    @Override
    public void register(Observer observer) {
        this.observer = observer;
    }
}

File: app/src/main/java/com/example/rickmortymvvm/util/observer/MutableObservableImpl.kt
package com.example.rickmortymvvm.util.observer

import com.example.rickmortymvvm.intrefaces.Observer
import com.example.rickmortymvvm.list.viewmodel.PresentationCharacterListAction
import com.example.rickmortymvvm.list.viewmodel.PresentationCharacterListState

class MutableObservableImpl : MutableObservable {
    private var observer: Observer? = null
    override fun update(state: PresentationCharacterListState) {
        observer!!.notify(state)
    }

    override fun update(action: PresentationCharacterListAction?) {
        observer!!.notify(action)
    }

    override fun register(observer: Observer?) {
        this.observer = observer
    }


}